src/savegame_json.cpp

```diff
@@ -334,13 +334,15 @@ void player_activity::deserialize( JsonIn &jsin )
     // this may cause inconvenience but should avoid any lasting damage to npcs
     if( is_obsolete || ( has_actor && ( data.has_null( "actor" ) || !data.has_member( "actor" ) ) ) ) {
         type = activity_id( "ACT_MIGRATION_CANCEL" );
+        actor = activity_actors::deserialize_functions.at( type )( jsin );
```

Comment:
This line is suspicious. `JsonIn` is a stream - once you read something it is gone. By this point `player_activity::deserialize()` has already consumed the entire activity and turned it into `JsonObject data`.
Currently `migration_cancel_activity_actor::deserialize()` does not read JsonIn and simply returns `migration_cancel_activity_actor().clone()` if in the future it does read JsonIn this may corrupt the stream.

https://github.com/CleverRaven/Cataclysm-DDA/blob/e49b6f7cc0b5038c8ca7ee940783b0ac07a77841/src/activity_actor.cpp#L2288-L2291
I would replace this line with:
```
actor = migration_cancel_activity_actor().clone(); 
```

Reply:
Oh yeah, I had a comment explaining why it's safe to give it jsin but forgot to push it.
I considered instantiating the actor directly but I'd need to include a large header file.
Alternatively, I could pass in a dummy JsonIn, but I couldn't find an easy way to make one.

Reply:
Alright, I added the extra include and instantiated the actor directly. It's cleaner.